KarhooUISDKUITests/Checkout/CheckoutSnapshotSpec.swift

```diff
@@ -0,0 +1,248 @@
+                        bookingMetadata: nil,
+                        router: MockCheckoutRouter(),
+                        loyaltyWorker: loyaltyWorker
+                    )
+                    sut.setupBinding(viewModel)
+                    
+                    let loyaltyUiModel = LoyaltyUIModel(
+                        loyaltyId: "loyaltyId",
+                        currency: "USD",
+                        tripAmount: 12.52,
+                        canEarn: true,
+                        canBurn: true,
+                        burnAmount: 150,
+                        earnAmount: 15, balance: 12345
+                    )
+                    navigationController.pushViewController(sut, animated: false)
+                    loyaltyWorker.modelSubject.send(.success(result: loyaltyUiModel))
+                    
+                }
+                it("Loyalty View should be visible") {
+                    testSnapshot(navigationController)
+                }
+            }
+            
+            context("when Checkout is opened with T&C checkbox required") {
+                beforeEach {
+                    KarhooTestConfiguration.isExplicitTermsAndConditionsConsentRequired = true
+                    quote = TestUtil.getRandomQuote(
+                        fleetName: "Fleet name",
+                        categoryName: "Category name",
+                        type: "type"
+                    )
+                    let journeyDetails = JourneyDetails.mockWithScheduledDate()
+                    viewModel = KarhooCheckoutViewModel(
+                        quote: quote,
+                        journeyDetails: journeyDetails,
+                        bookingMetadata: nil,
+                        router: MockCheckoutRouter()
+                    )
+                    sut.setupBinding(viewModel)
+                    navigationController.pushViewController(sut, animated: false)
+                }
+                it("checkbox should be visible") {
+                    testSnapshot(navigationController)
+                }
+            }
+            
+            context("when Checkout is opened with T&C checkbox required and user tap checkbox") {
+                beforeEach {
+                    KarhooTestConfiguration.isExplicitTermsAndConditionsConsentRequired = true
+                    quote = TestUtil.getRandomQuote(
+                        fleetName: "Fleet name",
+                        categoryName: "Category name",
+                        type: "type"
+                    )
+                    let journeyDetails = JourneyDetails.mockWithScheduledDate()
+                    viewModel = KarhooCheckoutViewModel(
+                        quote: quote,
+                        journeyDetails: journeyDetails,
+                        bookingMetadata: nil,
+                        router: MockCheckoutRouter()
+                    )
+                    sut.setupBinding(viewModel)
```

Comment:
Extract sut building & setup to designed method. It's called multiple times across the file.